app/src/main/java/pm/gnosis/heimdall/ui/tokens/TokensContract.kt

```diff
@@ -1,15 +1,19 @@
 package pm.gnosis.heimdall.ui.tokens
 
 import android.arch.lifecycle.ViewModel
-import io.reactivex.Flowable
 import io.reactivex.Observable
 import pm.gnosis.heimdall.common.util.Result
 import pm.gnosis.heimdall.data.repositories.model.ERC20Token
+import pm.gnosis.heimdall.ui.base.Adapter
 import java.math.BigInteger
 
 abstract class TokensContract : ViewModel() {
-    abstract fun observeTokens(): Flowable<List<ERC20Token>>
+    // If the address is not provided the current active account will be used
+    abstract fun observeTokens(refreshEvents: Observable<Unit>, ofAddress: BigInteger? = null): Observable<out Result<Adapter.Data<ERC20TokenWithBalance>>>
```

Comment:
why not use a setup method for the `ofAddress` (we did that before already, right?) Doesn't need to be async.